### ISSUE TITLE
Merge (most of) rustc_llvm into rustc_trans

### ISSUE THREAD

**Issue by rkruppe** [rust-lang/rust#41699]
After #41565 is merged, only the rustc_trans crate depends on LLVM. In #41473, @eddyb proposed eliminating the rustc_llvm crate altogether once that is achieved and merging it into rustc_trans, but @whitequark pointed out that this might regress rebuild times when trans changed and LLVM didn't. A better solution might be to make rustc_llvm a `*-sys` crate, i.e., it contains just the build script and the bare bones FFI definitions (`extern blocks` and types) while all the higher-level stuff moves into rustc_trans.

**Comment by rkruppe** [comment]
Closing in favor of #45274
